src/main/java/com/wishlist/services/ProductService.java

```diff
@@ -0,0 +1,52 @@
+package com.wishlist.services;
+
+import com.wishlist.exceptions.ProductNotFoundException;
+import com.wishlist.model.Product;
+import com.wishlist.repositories.ProductRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+@Transactional(readOnly = true)
```

Comment:
Transactional(readOnly = true) on top of the class makes all methods transactional (of course on the database level), but in the transactions no write is possible. Modifying methods should be annotated explicitly with Transational.